augur/augur_client.py

```diff
@@ -46,50 +46,73 @@ def __init__(self, hostname, port):
       return None
     market_info_dict = {}
     casting_map = {
+      'id': str,
+      'universe': str,
       'market_type': MarketInfo.Type,
       'min_price': Decimal,
       'max_price': Decimal,
       'cumulative_scale': Decimal,
-      'creation_time': datetime.fromtimestamp,
+      'author': str,
+      'creation_time': datetime,
       'creation_fee': Decimal,
       'settlement_fee': Decimal,
       'reporting_fee_rate': Decimal,
       'market_creator_fee_rate': Decimal,
       'market_creator_fees_balance': Decimal,
+      'market_creator_mailbox': str,
+      'market_creator_mailbox_owner': str,
       'initial_report_size': Decimal,
+      'category': str,
+      'tags': list,
       'volume': Decimal,
       'open_interest': Decimal,
       'outstanding_shares': Decimal,
       'reporting_state': ReportingState,
-      'end_time': datetime.fromtimestamp,
-      'finalization_time': datetime.fromtimestamp,
-      'last_trade_time': datetime.fromtimestamp,
+      'end_time': datetime,
+      'finalization_time': datetime,
+      'last_trade_time': datetime,
+      'description': str,
+      'details': str,
+      'scalar_denomination': str,
+      'designated_reporter': str,
       'designated_report_stake': Decimal,
+      'resolution_source': str,
       'num_ticks': Decimal,
       'tick_size': Decimal,
       'consensus': NormalizedPayout,
       'outcomes': OutcomeInfo,
     }
     for key, value in response_data.items():
       new_key = self._to_snake_case(key)
-      if value is None:
-        market_info_dict[new_key] = None
-      elif isinstance(value, (list, str)) and len(value) == 0:
-        market_info_dict[new_key] = value
-      else:
-        to_cast_type = casting_map.get(new_key, None)
-        if to_cast_type is None:
-          new_value = value
-        elif (to_cast_type is MarketInfo.Type or
-              to_cast_type is ReportingState):
-          new_value = to_cast_type[value.upper()]
-        elif to_cast_type is NormalizedPayout:
-          new_value = to_cast_type(*value)
-        elif to_cast_type is OutcomeInfo:
-          new_value = [OutcomeInfo(**x) for x in value]
+      to_cast_type = casting_map.get(new_key, None)
+      if not value:
+        if to_cast_type is list:
```

Comment:
This appears to be incorrect. Line 88 checks if ```value``` is ```False``` and if it is then...

1. Line 90 will iterate over all elements in ```value```, but ```value``` is ```False``` which means that it's either ```None``` or an empty ```list```, so iterating it will have no effect.

2. Line 92 will assign an empty string if ```not value```, but you already know that ```not value``` is ```True``` since you checked for that on line 88.

3. Line 94 assigns a result of ```None```, but what if ```value``` is of type ```bool``` and is ```False```, then you will assign a value of ```None``` to ```new_value``` when you should have assigned ```False```.

My suspicion is that the approach of having a giant dictionary with values with a callable object used to cast the value might have seemed like a good idea initially to save code, that it's becoming too error prone to use this method in practice.

I suspect that implementing this using the more direct approach will end up in less code written and code that is more easy to verify and reason about, ie:

```
  return MarketInfo(response_data['x'], Decimal(response_data['y']), datetime(response_data['z']), ...)
```